src/plugins/intel_cpu/src/nodes/multinomial.cpp

```diff
@@ -0,0 +1,176 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "multinomial.hpp"
+
+#include "ie_ngraph_utils.hpp"
+#include "openvino/op/multinomial.hpp"
+
+namespace ov {
+namespace intel_cpu {
+namespace node {
+
+Multinomial::Multinomial(const std::shared_ptr<ov::Node>& op, const GraphContext::CPtr& context)
+    : Node(op, context, NgraphShapeInferFactory(op, PortMask(NUM_SAMPLES_PORT))) {
+    std::string errorMessage;
+    if (!isSupportedOperation(op, errorMessage)) {
+        THROW_CPU_NODE_ERR(errorMessage);
+    }
+
+    auto multinomial_op = as_type_ptr<op::v13::Multinomial>(op);
+    m_with_replacement = multinomial_op->get_with_replacement();
+    m_global_seed = multinomial_op->get_global_seed();
+    m_log_probs = multinomial_op->get_log_probs();
+    m_op_seed = multinomial_op->get_op_seed();
+    m_output_precision = multinomial_op->get_convert_type();
+
+    for (size_t i = 0lu; i < op->get_input_size(); i++) {
+        if (is_type<op::v0::Constant>(op->get_input_node_ptr(i))) {
+            m_const_inputs[i] = true;
+        }
+    }
+}
+
+bool Multinomial::isSupportedOperation(const std::shared_ptr<const ov::Node>& op, std::string& errorMessage) noexcept {
+    try {
+        if (op->get_type_info() != op::v13::Multinomial::get_type_info_static()) {
+            errorMessage = "Only Multinomial operation from the opset13 is supported by the CPU plugin.";
+            return false;
+        }
+    } catch (...) {
+        return false;
+    }
+    return true;
+}
+
+void Multinomial::getSupportedDescriptors() {
+    if (getParentEdges().size() != 2) {
+        THROW_CPU_NODE_ERR("has incorrect number of input edges.");
+    }
+    if (getChildEdges().size() != 1) {
+        THROW_CPU_NODE_ERR("has incorrect number of output edges.");
+    }
+}
+
+void Multinomial::initSupportedPrimitiveDescriptors() {
+    m_probs_precision = getOriginalInputPrecisionAtPort(PROBS_PORT);
+    if (!one_of(m_probs_precision, ov::element::f32, ov::element::f16, ov::element::bf16)) {
+        m_probs_precision = ov::element::f32;
+    }
+
+    m_num_samples_precision = getOriginalInputPrecisionAtPort(NUM_SAMPLES_PORT);
+    if (!one_of(m_num_samples_precision, ov::element::i64, ov::element::i32)) {
```

Comment:
At this moment the plugin does not support int64, thus m_num_samples_precision can be int32 only.

Reply:
Removed!
Just out of curiosity, why do the tests pass still, for i64 precision covert?

Reply:
Looks like [convert precision transformation](https://github.com/openvinotoolkit/openvino/blob/master/src/common/transformations/src/transformations/convert_precision.cpp) should be added (as a follow up for this PR) to use i64 type. 
@nshchego Could you confirm?
But I'm also curious about the passing tests

Reply:
In general, that is required transformation.
@PiotrKrzem , could you please check that the model created in fn `MultinomialLayerTest::SetUp()` really has i64 input.

Reply:
I've checked, and it is indeed i64